### PR TITLE
Issue #13421: Add check to ensure all module setters have since in javadoc

### LINKED ISSUE #13421: Add `@since` in javadoc of each property setter
A requirement for implementing module properties macro
https://github.com/checkstyle/checkstyle/issues/13399

One of the 5 attributes of each property is the `since` attribute. In order for us to implement the properties macro, this information has to be stored somewhere in the class java file. Since we decided to use the setters of a module for generating the properties, the `since` has to be added in the property javadoc.

Screenshot from 2023-07-23 19-23-42

For example
```diff
    /**
     * Setter to allow single parameterless annotation to be located on the same line as
     * target element.
     *
     * @param allow User's value of allowSamelineSingleParameterlessAnnotation.
+     * @since 6.1
     */
    public final void setAllowSamelineSingleParameterlessAnnotation(boolean allow) {
        allowSamelineSingleParameterlessAnnotation = allow;
    }
```

### PR DESCRIPTION
**Pull request by stoyanK7**
Closes #13421 

---
DO NOT MERGE until all other PRs referencing #13421 are merged

### CODE REVIEW

**Comment by romani** [comment]
Please convert it for ready for review. Red CI will naturally block merge until final merge of javadoc update and rebase of this PR.

File: config/checkstyle-checks.xml
Comment (nrmancuso) [inline config/checkstyle-checks.xml:491]: Why any depth on `CLASS_DEF` and `METHOD_DEF`? Won't this also check nested classes unnecessarily?

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: config/checkstyle-checks.xml
Comment (stoyanK7) [reply config/checkstyle-checks.xml]: You're right. Done.

**Review by romani — APPROVED** [review]
ok to merge when CI is green

**Comment by stoyanK7** [comment]
There are some false positives such as `XpathFileGeneratorAstFilter` and `AbstractCheck`, `AbstractFileSetCheck`, etc..

I think the Xpath should be extended to not catch abstract classes.

**Comment by romani** [comment]
Let's keep focus on project tasks, optimization of validation can be done when all is working good.
We can create separate issue to optimize validation, suppression is fine for now.

**Review by nrmancuso — COMMENTED** [review]
@stoyanK7 please ping me when CI is green for final review

**Comment by romani** [comment]
PR is rebased by Github button

File: config/suppressions.xml
Comment (romani) [inline config/suppressions.xml:182]: `.... it is not Module property`

File: config/suppressions.xml
Comment (romani) [inline config/suppressions.xml:184]: this a good modules with properties, please share what is problem with them ?

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: config/suppressions.xml
Comment (stoyanK7) [reply config/suppressions.xml:184]: `fileContents` is not a property here. It is a setter used internally
https://checkstyle.org/filters/suppressioncommentfilter.html#Properties

Comment (romani) [reply config/suppressions.xml:184]: https://github.com/checkstyle/checkstyle/blob/63b4549b0c51924e266af2d60730911bc6358b86/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java#L586-L595

Comment (romani) [reply config/suppressions.xml:184]: can we use some nearby filters to suppress only this setter , not whole file ?

Comment (romani) [reply config/suppressions.xml:184]: please do:
```
 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java b/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
index 74d247e..8e23b8e 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
@@ -587,10 +587,8 @@ public class SuppressionCommentFilter
      * Set the FileContents for this filter.
      *
      * @param fileContents the FileContents for this filter.
-     * @noinspection WeakerAccess
-     * @noinspectionreason WeakerAccess - we avoid 'protected' when possible
      */
-    public void setFileContents(FileContents fileContents) {
+    private void setFileContents(FileContents fileContents) {
         fileContentsReference = new WeakReference<>(fileContents);
     }
 

```

Comment (stoyanK7) [reply config/suppressions.xml]: Done.

**Review by romani — APPROVED** [review]
ok to merge if CI pass

**Comment by stoyanK7** [comment]
@nrmancuso I added suppression for sevntu in https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/1023

**Comment by nrmancuso** [comment]
sevntu PR is merged, semaphore is restarted

**Review by nrmancuso — APPROVED** [review]
Good, let's merge as CI passes.

**Pull request merged**
